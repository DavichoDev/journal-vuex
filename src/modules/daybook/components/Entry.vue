<template>
    <div class="entry-container pointer mb-3 p-2"
         @click="$router.push({name: 'entry', params: {id: 10}})">
        <!-- Titulo -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">23</span>
            <span class="mx-1 fs-5">Octubre</span>
            <span class="mx-2 fw-light">2021, Sabado</span>
        </div>
        
        <div class="entry-description">
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Ducimus praesentium eius dicta optio, assumenda consequuntur ipsum eligendi reiciendis quod? Provident corrupti ut error a! Sapiente molestias dolorem sint ea deleniti.
        </div>
    </div>
</template>

<style lang="scss" scoped>
.pointer{ 
    cursor: pointer;
}
.entry-container{
    border-bottom: 1px solid #123e50;
    transition: 0.2 all ease-in;
    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2 all ease-in;
    }
    .entry-description{
        font-size: 12px;
    }
}
</style>