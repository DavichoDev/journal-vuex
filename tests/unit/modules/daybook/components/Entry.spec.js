import { shallowMount } from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue';
import { journalState } from '../../../mock-data/test-state';

describe('Entry.js Tests', () => {

    const mockRouter = {
        push: jest.fn()
    }

    let wrapper

    beforeEach(() => {
      wrapper = shallowMount( Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            } 
        }
      })
      jest.clearAllMocks()
    })

    test('Debe de pasar el snapshot', () => {
        expect( wrapper.html ).toMatchSnapshot()
    })

    test('Debe de redireccionar al hacer click con el entry-container', () => {
        const container = wrapper.find('.entry-container')
        const idEntry = wrapper.props().entry.id
        container.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({name: 'entry', params: {id: idEntry}})
    })

    test('pruebas en las propiedades computadas', () => {

        const day = wrapper.vm.day
        const month = wrapper.vm.month
        const yearDay = wrapper.vm.yearDay

        // TypeAssertions
        expect( typeof day ).toBe('number')
        expect( typeof month ).toBe('string')
        expect( typeof yearDay ).toBe('string')
        
        // ValueAssertions
        expect( day ).toBe(7)
        expect( month ).toBe('Noviembre')
        expect( yearDay ).toBe('2021, Domingo')
    })
})