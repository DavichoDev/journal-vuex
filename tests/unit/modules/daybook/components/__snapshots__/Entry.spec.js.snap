// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry.js Tests Debe de pasar el snapshot 1`] = `[Function]`;
